### PR TITLE
Add more information to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by tdcmeehan**
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,5 +1,50 @@
 # Contributing to Presto
 
+Thanks for your interest in Presto.  Our goal is to build a fast, scalable and reliable distributed SQL query engine for running interactive analytic queries against data sources of all sizes ranging from gigabytes to petabytes.
```
Comment (aweisberg) [inline CONTRIBUTING.md:3]: Our focus is no longer limited to interactive. We are clearly heavily into batch analytics for some time now :-P
```suggestion
Thanks for your interest in Presto.  Our goal is to build a fast, scalable and reliable distributed SQL query engine for running low latency interactive as well as batch analytic queries against data sources of all sizes ranging from gigabytes to petabytes.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,5 +1,50 @@
...
+Presto's [open issues are here](https://github.com/prestodb/presto/issues). Issues that would make a good first pull request for new contributors with the `beginner-task` tag. An easy way to get started helping the project is to *file an issue*. Issues can include bugs, new features, or documentation that looks outdated. For community support, [ask for help in Slack](https://join.slack.com/t/prestodb/shared_invite/enQtNTQ3NjU2MTYyNDA2LTYyOTg3MzUyMWE1YTI3Njc5YjgxZjNiYTgxODAzYjI5YWMwYWE0MTZjYWFhNGMwNjczYjI3N2JhM2ExMGJlMWM).
+
+## Contributions
+
+Presto welcomes contributions from everyone.
+
+Contributions to Presto should be made in the form of GitHub pull requests. Each pull request will
+be reviewed by a contributor or [committer](https://github.com/prestodb/presto/wiki/committers) in the project.
+Large contributions should have an associated Github issue.
+
```
Comment (aweisberg) [inline CONTRIBUTING.md:16]:
```suggestion

Code review is also form of contribution and it is actually one of the most valuable. All code has to be reviewed in order to get merged and the more reviewers we have the better the review experience for everyone.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,5 +1,50 @@
...
+## Pull Request Checklist
+
+- Branch from the master branch and, if needed, rebase to the current master
+  branch before submitting your pull request. If it doesn't merge cleanly with
+  master you may be asked to rebase your changes.
+
+- Ensure that each commit is correct independently (i.e., each commit should compile and pass tests).
```
Comment (aweisberg) [inline CONTRIBUTING.md:23]: It seems like this should delegate to the review and commit guidelines to reduce duplication?

**Review by aweisberg — CHANGES_REQUESTED** [review]
Thanks for updating this. Some suggestions.

File: CONTRIBUTING.md (hunk 3 above)
Comment (tdcmeehan) [reply CONTRIBUTING.md:23]: I think it's useful to highlight the important stuff as bullets, as `CONTRIBUTING.md` should be more easily discoverable than the wiki guidelines, and this is more focused to the contributors than committers.  How about I rework it to something like: 

```
- Make sure you follow the [review and commit guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines), in particular:

  - Ensure that each commit is correct independently (i.e., each commit should compile and pass tests).
    When possible, the size of the commit should be reduced for ease of review, while still ensuring
    that it is independently correct.

  - Make sure commit messages [follow these guidelines](https://chris.beams.io/posts/git-commit/).  In particular (from the guidelines):

    * Separate subject from body with a blank line
    * Limit the subject line to 50 characters
    * Capitalize the subject line
    * Do not end the subject line with a period
    * Use the imperative mood in the subject line
    * Wrap the body at 72 characters
    * Use the body to explain what and why vs. how
    ```

File: CONTRIBUTING.md (hunk 1 above)
Comment (tdcmeehan) [reply CONTRIBUTING.md:3]: Tweaked it to be similar to that.

File: CONTRIBUTING.md (hunk 2 above)
Comment (aweisberg) [reply CONTRIBUTING.md:16]: I think that doesn't go far enough to set expectations. I think we generally don't do a good job at this as a project and the result is that committers do all the review. Throughout the lifecycle of someones escalating participation in the project I think it needs to be clear that they should be reviewing as much code as they are contributing.

It also clues people into why their review might not be done immediately.

This really isn't something I feel strongly about in this specific instance, but in general I think we are struggling with this as a project.

File: CONTRIBUTING.md (hunk 3 above)
Comment (aweisberg) [reply CONTRIBUTING.md]: SGTM

**Review by aweisberg — APPROVED** [review]

**Comment by rohanpednekar** [comment]
Thanks, @tdcmeehan for doing this, added few comments, otherwise LGTM. 👍

**Comment by nmahadevuni** [comment]
@rohanpednekar @aweisberg : Are these changes going into https://prestodb.io/community.html page? Because the CLA link there is broken.

**Comment by aweisberg** [comment]
I think we should remove that link and just link to CONTRIBUTING.md

File: CONTRIBUTING.md (hunk 2 above)
Comment (tdcmeehan) [reply CONTRIBUTING.md:16]: That's a good point.  How about this:

```
## Contributions

Presto welcomes contributions from everyone.

Contributions to Presto should be made in the form of GitHub pull request submissions and reviews. 

Each pull request submission will be reviewed by a contributor or [committer](https://github.com/prestodb/presto/wiki/committers) 
in the project. Only committers may merge a pull request. Large contributions should have an associated Github issue.

Pull request reviews are encouraged for anyone in the community who would like to contribute to Presto, and are
expected from contributors and committers in at least equal proportion to their code contributions.
```

**Review by rohanpednekar — APPROVED** [review]

**Review by rohanpednekar — APPROVED** [review]

**Review by highker — APPROVED** [review]

**Pull request merged**